src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -812,101 +731,204 @@ public int getCurrentRunningCount() {
+                 * now is to introduce locking.
+                 */
+                // must get executeReference before time is checked
+                short executeReference = nextTask.getExecuteReference();
+                long taskDelay = nextTask.getScheduleDelay();
+                if (taskDelay > 0) {
+                  if (taskDelay == Long.MAX_VALUE) {
+                    // TODO - this will cause a spin if there is only one recurring task
+                    // don't sleep forever, this task is just starting to execute, so spin
+                    Thread.yield();
+                  }
+                  if (queued) {
+                    if (! workerTimedPark) {
+                      // we can only park after we have queued, then checked again for a result
+                      workerTimedPark = true;
+                      LockSupport.parkNanos(Clock.NANOS_IN_MILLISECOND * taskDelay);
+                      workerTimedPark = false;
+                      continue idle;
+                    } else {
+                      // there is another worker already doing a timed park, so we can wait till woken up
+                      LockSupport.park();
+                      continue idle;
+                    }
+                  } else {
+                    addWorkerToIdleChain(worker);
+                    queued = true;
+                  }
+                } else if (nextTask.canExecute(executeReference)) {
+                  return nextTask;
+                }
+              }
+              // reset interrupted status if we may block and have not checked
+              if (queued && ! interruptedChecked) {
+                interruptedChecked = true;
+                if (Thread.interrupted()) {
+                  // verify we were not interrupted due to pool shutdown
+                  continue idle;
+                }
+              }
+            } // end pollTask loop
+          }
+        } // end idle loop
+      } finally {
+        // if queued, we must now remove ourselves, since we are either shutdown, or have a task to run
+        if (queued) {
+          // only if we are the head of the queue can we remove ourselves without locking
+          if (idleWorker.get() == worker && 
+              idleWorker.compareAndSet(worker, worker.nextIdleWorker)) {
+            // no further actions needed, change occurred in if condition
+          } else {
+            /* I wish we could remove this lock...
+             * Right now we must lock because we must atomically update the worker holding our 
+             * reference to point to the worker held inside our reference.  This means two 
+             * references must be updated in a single operation
+             */
+            /* Locking operations here don't interfere with concurrent operations on idleWorker's 
+             * atomic reference (which is why that was checked first)
+             */
+            synchronized (idleWorker) {
```

Comment:
You know... this atomicReference linklist could probably be commonized into a pretty interesting concurrent structure, then  you could move these operations to calls on it to help reduce some of the code in this class a little.